File: count_and_say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return "1"
        toCount = self.countAndSay(n-1)
        prev = toCount[0]
        frequency = 1
        ans = ""
        for i in range (1, len(toCount)):
            if prev == toCount[i]:
                frequency += 1
            else:
                ans += str(frequency) + str(prev)
                prev = toCount[i]
                frequency = 1
        return ans + str(frequency) + str(prev)

sol = Solution().countAndSay(4)
print(sol)

File: count_and_say.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func countAndSay(n int) string {
	if n == 1 {
		return "1"
	}
	var toCount = []rune(countAndSay(n - 1))
	var prev rune = toCount[0]
	var frequency int = 1
	var ans strings.Builder
	i := 1
	for i < len(toCount) {
		if prev == toCount[i] {
			frequency++
		} else {
			ans.WriteString(strconv.Itoa(frequency))
			ans.WriteRune(prev)
			frequency = 1
			prev = toCount[i]
		}
		i++
	}
	ans.WriteString(strconv.Itoa(frequency))
	ans.WriteRune(prev)
	return ans.String()

}

func main() {
	k := countAndSay(4)
	fmt.Println(k)
}
